locator/abstract_replication_strategy.cc

```diff
@@ -136,7 +147,7 @@ insert_token_range_to_sorted_container_while_unwrapping(
 }
 
 dht::token_range_vector
-effective_replication_map::do_get_ranges(noncopyable_function<bool(inet_address_vector_replica_set)> should_add_range) const {
+effective_replication_map::do_get_ranges(noncopyable_function<std::optional<bool>(const inet_address&)> should_add_range) const {
```

Comment:
std::optional<bool> is just too complicated, you can't understand what the different levels mean.

Reply:
okay, it can be simplified by returning stop_iteration and getting a `bool&` for the result as a parameter.